models/LogReg.py

```diff
@@ -0,0 +1,167 @@
+# For hyperopt (parameter optimization)
+from hyperopt import STATUS_OK
+# sklearn models
+from sklearn import linear_model
+# diagnostics
+from sklearn.metrics import balanced_accuracy_score
+from scripts.utils import run_hyperopt
+import joblib
+
+
+class LogReg:
+    '''
+    Methods for deploying sklearn's logistic regression
+    implementation with hyperparameter optimization.
+    Data agnostic (i.e. user supplied data inputs).
+    TODO: Currently only supports binary classification.
+        Add multinomial functions and unit tests.
+        Add functionality for regression(?)
+    Inputs:
+    params: dictionary of logistic regression input functions.
+        keys max_iter, tol, and C supported.
+    random_state: int/float for reproducible intiailization.
+    '''
+
+    # only binary so far
+    def __init__(self, params=None, random_state=0):
+        keys = ['max_iter', 'tol', 'C']
+        # defaults to a fixed value for reproducibility
+        self.random_state = random_state
+        # dictionary of parameters for logistic regression model
+        self.params = params
+        if self.params is None:
+            self.model = linear_model.LogisticRegression(
+                            random_state=self.random_state
+                        )
+        else:
+            if all(key in params.keys() for key in keys):
```

Comment:
I'm not sure if this is the most correct/robust way to do this.  The `LogisticRegression` model has defaults for these parameters, so it may be OK if some are missing.  You just need to make sure they exist if you want to pass them along.  Right now, you only allow 0 parameters or all 3 parameters, but maybe it's OK for just 1 or 2?

One way to manage this is with the `**kwargs` object that you can pass through, perhaps?

Reply:
This is my first time using `**kwargs` but I saw a recommendation to use `kwargs.pop('key', default_value)` to pull option args from the input. This system should support any combination of input parameters, including ones that are not supported. I have updated the `__init__` and its relevant unit test. Let me know if you have feedback!